storage-node/src/cache/mod.rs

```diff
@@ -9,14 +20,35 @@ pub type ParpulseCacheKey = String;
 /// This is just a prototype and we might refine it later.
 pub type ParpulseCacheValue = (String, usize);
 
-/// [`ParpulseCache`] offers a unified interface for different cache algorithms.
-///
-/// Feel free to modify the methods.
+impl CacheValue for ParpulseCacheValue {
+    type Value = String;
+
+    fn into_value(self) -> String {
+        self.0
+    }
+
+    fn as_value(&self) -> &String {
+        &self.0
+    }
```

Comment:
```suggestion
    fn into_value(self) -> Self::Value {
        self.0
    }

    fn as_value(&self) -> &Self::Value {
        &self.0
    }
```

Wouldn't this make more sense? Other than this, LGTM.